File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string answer;
        int i = 0, j = 0;
        while (i < word1.size() || j < word2.size()) {
            if (i < word1.size()) {
                answer.push_back(word1[i]);
                i++;
            }
            if (j < word2.size()) {
                answer.push_back(word2[j]);
                j++;
            }
        }
        return answer;
    }
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        answer = []
        i, j = 0, 0
        while i < len(word1) or j < len(word2):
            if i < len(word1):
                answer.append(word1[i])
                i += 1
            if j < len(word2):
                answer.append(word2[j])
                j += 1
        return ''.join(answer)